src/client/index.ts

```diff
@@ -538,10 +540,24 @@ export class Client {
 			this.on("token-proof", null, { data, issuer, error });
 		},
 		emitErrorToClient: (error: Error, issuer = "none") => {
+
+			this.checkInternetConnectivity();
+
 			this.on("error", null, {error, issuer});
 		}
 	};
 
+	checkInternetConnectivity(): void {
+		if (!navigator.onLine) {
+			if (this.config.type === 'active') {
+				setTimeout(() => {
+					this.ui.showError(this.config.noInternetErrorMessage ?? NO_INTERNET_ERROR_MESSAGE);
+				}, 1000);
+			}
+			throw new Error(this.config.noInternetErrorMessage ?? NO_INTERNET_ERROR_MESSAGE)
+		}
+	}
```

Comment:
I think we should add a return boolean value to this function. This so we can use this inside the logic to stop the following action from triggering if there is no connection.